### DISCUSSION TITLE
How to exit the v2 dev server quickly on SIGTERM?

### DISCUSSION THREAD

**Discussion by ngbrown** [remix-run/remix#6956] [Q&A] [answered]
I'm running the remix dev server (v2) in a container with `docker compose` for various reasons. When the container is being shutdown or restarted, `SIGTERM` isn't being handled by Remix, so Docker waits until a timeout to kill the container. The application `server.js` is handling the `SIGTERM` and calling `process.exit(0)`, but it's just a child process of the Remix dev server.

#2528 appears to have added this to the `remix-serve` code, but I can't find any similar implementation in the v2 dev server code.

The `serve()` function below returns a `Promise` that doesn't seem to ever be resolved or rejected. Specifically, the `finally` callback never runs (I added `console.log()` to check). What is the source/reason for this pattern?

https://github.com/remix-run/remix/blob/97d82db0556ba8b40518898a75d6838e0bf3ced1/packages/remix-dev/devServer_unstable/index.ts#L267-L272

If I change the `Promise` slightly, I can get the dev server to quickly exit because the `finally` does execute:

```ts
return new Promise((resolve, reject) =>
  ["SIGINT", "SIGTERM"].forEach((signal) =>
    process.once(signal, (code) => reject(code))
  )
).finally(async () => {
  // ...
});
```

Is this is on the right track?

**Comment by ngbrown** [comment]
Should I open a pull request with this change in the dev server returned `Promise`?

**Reply by cliffordfajardo** [reply]
Is this something you are still running into with the latest Remix version?

**Reply by ngbrown** [reply]
I'm still locally patching @remix-run/dev v1.19.3. @cliffordfajardo are you having a similar problem or do you know of a fix being in the code base?

**Comment by MichaelDeBoey** [comment]
CC/ @pcattori

**Comment by pcattori** [comment, marked as answer]
Should be fixed by https://github.com/remix-run/remix/pull/7280
